src/utils/material.js

```diff
@@ -19,9 +19,11 @@ module.exports.updateMap = function (shader, data) {
 
   // Texture removed.
   if (!material.map) { return; }
+  shader.textureSrc = null;
```

Comment:
can you say more?  the shader is using .textureSrc as a shadow to avoid redundant changes, and it wasn't being removed / reset when the texture was removed/reset.